src/parser/deserializer.ts

```diff
@@ -445,12 +443,14 @@ function deserializeObject(
           _buffer[i] = buffer[index + i];
         }
 
-        if (promoteUUIDs && subType === 4) {
-          value = new Binary(_buffer, subType).toUUID();
-        } else if (promoteBuffers && promoteValues) {
+        if (promoteBuffers && promoteValues) {
           value = _buffer;
         } else {
           value = new Binary(_buffer, subType);
+          value = new Binary(buffer.slice(index, index + binarySize), subType);
+          if (subType === constants.BSON_BINARY_SUBTYPE_UUID_NEW) {
+            value = value.toUUID();
+          }
```

Comment:
Another consistency comment, lets use either an if stmt or a ternary in both places, really minor but might as well have the code match up

Reply:
I added an if else since the ternary would be a bit long / hard to read!